tests/Tutor.Infrastructure.Tests/Unit/EventStore/QueryInterfaceTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Shouldly;
+using Tutor.Infrastructure.Database.EventStore;
+using Tutor.Infrastructure.Tests.TestData.EventStore;
+using Xunit;
+
+namespace Tutor.Infrastructure.Tests.Unit.EventStore
+{
+    public class QueryInterfaceTests : IClassFixture<EventStoreManager>
+    {
+        private readonly IEventStore _eventStore;
+
+        public QueryInterfaceTests(EventStoreManager eventStoreManager)
+        {
+            _eventStore = eventStoreManager.EventStore;
+        }
+
+        [Theory]
+        [MemberData(nameof(QueryTestData.GetQueries), MemberType = typeof(QueryTestData))]
+        public void After_before_conditions(Query parameters)
```

Comment:
There should probably be a better way to test different orders of method calls but I couldn't think of one.

Reply:
Implemented in 125b47c6a721df6ecd9e083acde1a46321a70a8f.